### ISSUE TITLE
bug(`anvil`): Anvil panics and crashes when attempting to load txs by block via Otterscan API (`OTS`)

### ISSUE THREAD

**Issue by akshatmittal** [foundry-rs/foundry#8822]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (ea3ba89 2024-09-05T00:28:03.529093300Z)

### What command(s) is the bug in?

`anvil --fork-url https://eth.public-rpc.com`

### Operating System

Windows

### Describe the bug

Attempting to read transactions from a block via the Otterscan API kills `anvil` with the following error:

```
eth_getTransactionReceipt
thread 'tokio-runtime-worker' panicked at crates\anvil\src\eth\otterscan\api.rs:412:73:
called `Option::unwrap()` on a `None` value
```

Reproduction:

1. Run `anvil --fork-url https://eth.public-rpc.com`
2. Visit [Otterscan for any block](https://otterscan-devnet.metacrypt.org/block/20687632/txs) ([source](https://github.com/0xMetacrypt/otterscan-devnet))

Expected:

Should not crash. (Probably acceptable to return no transactions if block before fork block, ideally not)

**Comment by mattsse** [comment]
which ots method are you calling?

**Comment by akshatmittal** [comment]
I believe it's `ots_getBlockTransactions` but if you open the link you'll see all the requests in the console.

**Comment by yash-atreya** [comment]
This should be fixed with #8835
